routing/pathfind.go

```diff
@@ -37,6 +37,11 @@ const (
 	// to avoid resizing and copies. It should be number on the same order as
 	// the number of active nodes in the network.
 	estimatedNodeCount = 10000
+
+	// hopHintCapacity is the capacity we assume for hop hint channels. This
+	// is a high number, which expresses that a hop hint should be able to
+	// route payments.
+	hopHintCapacity = btcutil.Amount(10_0000_0000)
```

Comment:
I am not sure about this... Many times the node receiving the payment will give you multiple hop hints because none of them can fully handle the payment amount.

Reply:
From outbound communication:

> so the observation is that in the default configuration, the bimodal algo likes to split a lot if it doesn't have any other chances (this is natural and the best strategy if you assume unbalanced channels that have liquidity in your disfavor). increasing the scale parameter leads to less splitting

Couple of things:

- Why do we split the 0s in groups of four? `10_0000_0000`? 

- Any chances of using the invoice amount for the first try and split if needed from there?

Reply:
I think that's the reason we want to use `hopHintCapacity`. Based on the PR description, previously we won't factor the failure from using hop hints as there's no capacity info, and we can do that now as we are using a fake one. This means we'd only try one of the multiple hop hints of the payment previously and now we will try more.

nit: maybe name it `fakeHopHintCapacity`, `assumedHopHintCapacity` or something similar to make it more obvious that there's no capacity info in the hop hint?

Reply:
> I am not sure about this... Many times the node receiving the payment will give you multiple hop hints because none of them can fully handle the payment amount.
This was explained by @yyforyongyu's comment. The number is just high enough that all hop hint channels would be tried and that the probability estimation can be used (to factor in previously failed channels).

> * Why do we split the 0s in groups of four? `10_0000_0000`?
This was just to indicate that we use a high BTC amount. The value is quite arbitrary, but I was not sure how the calculation would behave with very high numbers. Essentially I wanted to make sure that this number is higher than any payment amount, but being low enough not to introduce numerical inaccuracy. The amount is open for debate as well, I'll test this a bit more. We could also use `BimodalScaleMsatMax`, but I thought a clear separation of concerns was better.

> Any chances of using the invoice amount for the first try and split if needed from there?
If we would use the invoice amount (or current split payment amount) for the capacity, then we would not try the channel, because the model also includes the effect of a penalty when the channel is exactly exhausted (virtually).

Reply:
I think conceptually the hop hint capacity was infinite previously, but now it's capped. A bizarre case is, what if the payment is even over 10 btc? Maybe for non-wumbo channels, we can use that max channel size here. For wumbo channels, sky is the limit☄️ but we can at least start with 10 btc and see the effect. If needed maybe make it configurable someday.

Reply:
Right, not really sure about that magic number. How would one get the wumbo flag here, a hop hint doesn't [include it](https://github.com/lightning/bolts/blob/master/11-payment-encoding.md#tagged-fields)? One could query the graph for the short chan id, but given that hop hints are mostly private channels, that's probably not helpful, it also adds query latency.

Reply:
> Any chances of using the invoice amount for the first try and split if needed from there?

We could also use `amount * 10` (the current shard amount) as the capacity, that way we would get rid of the bound and it seems less magical. However, there's some slight interaction for small amounts with regards to MPP, see here these lines: https://github.com/lightningnetwork/lnd/blob/3967fb9340989de2980358eb0390f4991a65ff66/routing/probability_bimodal.go#L485-L499. So essentially we would gradually also reduce the fail amount when splitting, adjusting it to `amount * 10`.